File: docs/docs/API-Reference/python-examples/workflows-api/example-quickstart-background-poll.py
import os
import time

import requests

base = os.environ.get("LANGFLOW_URL") or os.environ.get("LANGFLOW_SERVER_URL", "")
flow_id = os.environ.get("FLOW_ID", "")
api_key = os.environ.get("LANGFLOW_API_KEY", "")

headers = {"Content-Type": "application/json", "x-api-key": api_key}

start = requests.post(
    f"{base}/api/v2/workflows",
    headers=headers,
    json={
        "flow_id": flow_id,
        "input_value": "Process this in the background",
        "session_id": "session-456",
        "mode": "background",
    },
    timeout=60,
)
start.raise_for_status()
job_id = start.json()["job_id"]
print(f"Queued job {job_id}")

while True:
    status = requests.get(
        f"{base}/api/v2/workflows",
        headers=headers,
        params={"job_id": job_id},
        timeout=60,
    )
    status.raise_for_status()
    body = status.json()

    if body.get("object") == "response" and body.get("status") == "completed":
        print(body["output"]["text"])
        break

    if body.get("status") in {"failed", "cancelled", "timed_out"}:
        raise SystemExit(f"Job ended with status {body.get('status')}")

    time.sleep(1)

File: docs/versioned_docs/version-1.12.0/API-Reference/javascript-examples/workflows-api/example-quickstart-background-poll.js
(async () => {
  const base = (process.env.LANGFLOW_URL ?? process.env.LANGFLOW_SERVER_URL ?? "").replace(/\/$/, "");
  const flowId = process.env.FLOW_ID ?? "";
  const apiKey = process.env.LANGFLOW_API_KEY ?? "";

  const headers = {
    "Content-Type": "application/json",
    "x-api-key": apiKey,
  };

  const start = await fetch(`${base}/api/v2/workflows`, {
    method: "POST",
    headers,
    body: JSON.stringify({
      flow_id: flowId,
      input_value: "Process this in the background",
      session_id: "session-456",
      mode: "background",
    }),
  });

  if (!start.ok) {
    throw new Error(`HTTP ${start.status}: ${await start.text()}`);
  }

  const { job_id: jobId } = await start.json();
  console.log(`Queued job ${jobId}`);

  while (true) {
    const status = await fetch(`${base}/api/v2/workflows?job_id=${encodeURIComponent(jobId)}`, {
      headers,
    });
    if (!status.ok) {
      throw new Error(`HTTP ${status.status}: ${await status.text()}`);
    }

    const body = await status.json();
    if (body.object === "response" && body.status === "completed") {
      console.log(body.output.text);
      break;
    }

    if (["failed", "cancelled", "timed_out"].includes(body.status)) {
      throw new Error(`Job ended with status ${body.status}`);
    }

    await new Promise((resolve) => setTimeout(resolve, 1000));
  }
})().catch((error) => {
  console.error(error);
  process.exit(1);
});
